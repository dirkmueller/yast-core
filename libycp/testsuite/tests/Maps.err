Parsed:
----------------------------------------------------------------------
$[]
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
$[]
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
$["/home":3200, "/usr":560]
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
$[1:true, "string":(8 + 9)]
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
$[1:true, 2:[true, false], "string":(8 + 9)]
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
$[`x:true, `y:false]
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
  string `s = "s";
  integer `i = 1;
  map `m = $[i:i, s:s];
  return m;
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
  map `m1 = $[1:"1", 2:"2", 3:"3"];
  map `m2 = $[3:"4", 4:"4", 5:"5"];
  map `m = union (m1, m2);
  if ((size (m) != 5)) {
  return false;
}
  if ((lookup (m, 3, nil) != "4")) {
  return false;
}
  return true;
}
----------------------------------------------------------------------
[libycp] YCPMap.cc(add):52 Only integer, string, or symbol constant allowed as key in map
[Scanner] tests/Maps.ycp:41 Syntax error: Wrong or missing element at start of map
