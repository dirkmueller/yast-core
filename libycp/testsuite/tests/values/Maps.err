Parsed:
----------------------------------------------------------------------
$[]
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
$[]
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
$["/home":3200, "/usr":560]
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
$[1:true, "string":(8 + 9)]
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
$[1:true, 2:[true, false], "string":(8 + 9)]
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
$[`x:true, `y:false]
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // string s
    // integer i
    // map m
    // filename: "tests/values/Maps.ycp"
    string s = "s";
    integer i = 1;
    map m = $[s:s, i:i];
    return m;
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // map m1
    // map m2
    // map m
    // filename: "tests/values/Maps.ycp"
    map m1 = $[1:"1", 2:"2", 3:"3"];
    map m2 = $[3:"4", 4:"4", 5:"5"];
    map m = union (m1, m2);
    if ((size (m) != 5))
    return false;
    if ((m[3]:nil != "4"))
    return false;
    return true;
}
----------------------------------------------------------------------
[Parser] tests/values/Maps.ycp:41 Bad type for key
Parsed:
----------------------------------------------------------------------
Parser error
----------------------------------------------------------------------
[Interpreter] tests/values/Maps.ycp:41 Parser error
