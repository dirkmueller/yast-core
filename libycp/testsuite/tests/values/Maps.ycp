
# ---------------------------------------------------------
#
#  Filename:    Maps.ycp
#
#  Purpose:     test cases for YCP test
#
#  Creator:     jd@suse.de
#
#  Maintainer:  jd@suse.de
#
# ---------------------------------------------------------


# maps as associative arrays

$[ ]
$[]
$[ "/usr": 560, "/home":3200 ]
$[1: true, "string": 8+9 ]
$[1: true, 2: [ true, false ], "string": 8+9 ]
$[`x : true, `y : false ]

{ string s = "s"; integer i = 1; map m = $[ s:s, i:i ]; return m; }

{
    map m1 = $[1:"1", 2:"2", 3:"3"];
    map m2 = $[3:"4", 4:"4", 5:"5"];

    map m = union (m1, m2);

    if (size (m) != 5)
	return false;
    if (lookup (m, 3, nil) != "4")
	return false;
    return true;
}

# wrong examples, must be at end of file !!

$[ true: false, nil : "" ]
