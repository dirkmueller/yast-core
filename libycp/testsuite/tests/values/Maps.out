($[])
($[])
($["/home":3200, "/usr":560])
($[1:true, "string":17])
($[1:true, 2:[true, false], "string":17])
($[`x:true, `y:false])
($[1:1, "s":"s"])
(true)
